### PR TITLE
Doc 4340: Add CorDapp Packaging tutorial content

### PR DESCRIPTION
**Pull request by amiegrace12**

### CODE REVIEW

**Review by paulmoloneyr3 — CHANGES_REQUESTED** [review]

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md [hunk 1]
```diff
@@ -1,30 +1,30 @@
...
-    identifier: corda-5-alpha-tutorial-dev-one
-    weight: 1000
+    parent: corda-5-alpha-tutorials-deploy
+    identifier: corda-5-alpha-tutorial-deploy-cordapp-packaging
+    weight: 2000
 section_menu: corda-5-alpha
 ---
 
-This section describes how to package your CorDapp as CPKs, CPBs, and CPIs. You can read more about the CorDapp packaging format in the [Key concepts](../../introduction/key-concepts.html#packaging) section.
+This section describes how to package your CorDapp as CPKs, CPBs, and CPIs.
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md:12]: I think we could leave the link in?

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
 
 ## Before You Start
 
-You must install [Corda CLI](../corda-cli/overview.html).
+You will need a working copy of `corda-cli` with the MGM and package plugins installed.
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md:16]:
```suggestion
You will need a working copy of the Corda CLI with the MGM and package plugins installed. 
```
Also, may not need to specify "with the MGM and package plugins installed" as I think they are now installed by default..

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md [hunk 3]
```diff
@@ -1,30 +1,30 @@
...
 ## Before You Start
 
-You must install [Corda CLI](../corda-cli/overview.html).
+You will need a working copy of `corda-cli` with the MGM and package plugins installed. 
+<!-- See the [Corda CLI Plugin Host](https://github.com/corda/corda-cli-plugin-host#setupbuild). -->
 
-<!--## Configure the plugin in gradle???
+## Configuring the Gradle Plugin
 
 This describes how to convert an existing CorDapp project to the new Gradle plugin.
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md:21]:
```suggestion
To convert an existing CorDapp project to the new Gradle plugin:
```

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md [hunk 4]
```diff
@@ -33,27 +33,27 @@ This describes how to convert an existing CorDapp project to the new Gradle plug
...
     }
     ```
-1. Add the plugin to the plugins section of `settings.gradle`:
+3. Add the plugin to the plugins section of `settings.gradle`:
     ```groovy
     id 'net.corda.plugins.cordapp-cpk2' version cordaGradlePluginsVersion2
     id 'net.corda.plugins.cordapp-cpb2' version cordaGradlePluginsVersion2
     ```
-1. Inside the cordapp project change the plugins block at the top of the file:
+4. Inside the cordapp project, change the plugins block at the top of the file:
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md:41]:
```suggestion
4. Inside the CorDapp project, change the plugins block at the top of the file:
```

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md [hunk 5]
```diff
@@ -33,27 +33,27 @@ This describes how to convert an existing CorDapp project to the new Gradle plug
...
+## Generating a Code Signing Key
 
 To generate a code signing key for signing the CPI:
 
 1. Generate a signing key:
     ```shell
     keytool -genkeypair -alias "signing key 1" -keystore signingkeys.pfx -storepass "keystore password" -dname "cn=CPI Plugin Example - Signing Key 1, o=R3, L=London, c=GB" -keyalg RSA -storetype pkcs12 -validity 4000
     ```
-2. If you are using the Gradle plugin default signing key, you must import it into our key store. Save the following text into a file named `gradle-plugin-default-key.pem`:
+2. If you are using the default Gradle plugin signing key, you must import it into our key store. Save the following text into a file named `gradle-plugin-default-key.pem`:
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md:56]: not sure about the "our"..

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md [hunk 6]
```diff
@@ -79,18 +79,24 @@ This key can be generated once and kept for reuse.
 
 ## Generating a Group Policy File
 
-To generate a group policy file:
+If you intend to run a basic static network setup, you can use the `corda-cli` MGM plugin to generate a group policy file. For example, you could run the following command to generate the file:
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md:82]:
```suggestion
If you intend to run a basic [static network](../network-types.html#static-networks) setup, you can use the Corda CLI to generate a group policy file. For example, you could run the following command to generate the file:
```

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md [hunk 7]
```diff
@@ -79,18 +79,24 @@ This key can be generated once and kept for reuse.
...
 
-The gradle plugin builds the CPB. To create a CPB for a CPI:
-```shell
+Refer to the [Group Policy](../../../../../../en/platform/corda/5.0-alpha/deploying/group-policy.html) section to learn more about the `GroupPolicy.json` file.
+
+<!-- For more information on this plugin, refer to the [README.md](https://github.com/corda/corda-runtime-os/blob/release/os/5.0/tools/plugins/mgm/README.md). -->
+
+If you are running a dynamic network, you will need to export the group policy file from the MGM using the HTTP API. To learn how to do this, see the [MGM Onboarding](../deployment-tutorials/onboarding/mgm-onboarding.html) tutorial for more information.
+
+## Building a Version 2 CPI
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md:93]:
```suggestion
## Building a CPI
```

**Review by nadinequinr3 — CHANGES_REQUESTED** [review]

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md [hunk 8]
```diff
@@ -1,30 +1,30 @@
...
 
 1. Add a new version number to `gradle.properties`:
     ```groovy
     cordaGradlePluginsVersion2=7.0.0-SNAPSHOT
     ```
-1. Add this repository to pluginManagement/repositories in `settings.gradle`:
+2. Add this repository to pluginManagement/repositories in `settings.gradle`:
     ```groovy
     maven {
         url "${artifactoryContextUrl}/corda-dev"
```
Comment (josephzunigadaly) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md:30]: Should we delete this and suggest cloning the CSDE template repo as a starting point?

Comment (josephzunigadaly) [reply content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md:30]: The CSDE template is already configured but I guess it is handy to have standalone instructions for when we need them.

Comment (josephzunigadaly) [reply content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md:30]: The version number is now "7.0.1-SNAPSHOT" in the git repo.

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md [hunk 9]
```diff
@@ -79,18 +79,24 @@ This key can be generated once and kept for reuse.
...
 ## Building a CPI
 
-The gradle plugin builds the CPB. To create a CPB for a CPI:
-```shell
+The Gradle plugin builds the CPB. Run the following command to turn a CPB into a CPI:
+```shell 
 ./corda-cli.sh package create-cpi \
     --cpb mycpb.cpb \
-    --group-policy GroupPolicy.json \
+    --group-policy TestGroupPolicy.json \
```
Comment (josephzunigadaly) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md:99]: The filename should match the export command on line 84. So GroupPolicy.json was correct.

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md (hunk 5 above)
Comment (amiegrace12) [reply content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md:56]: Should it be your?

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md (hunk 8 above)
Comment (nadinequinr3) [reply content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md:30]: The CSDE is not included in the Alpha release

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md (hunk 2 above)
Comment (amiegrace12) [reply content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md:16]: Will I take "with the MGM and package plugins installed" out?

**Review by nadinequinr3 — APPROVED** [review]

**Review by paulmoloneyr3 — DISMISSED** [review]

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md [hunk 10]
```diff
@@ -1,30 +1,30 @@
...
-    identifier: corda-5-alpha-tutorial-dev-one
-    weight: 1000
+    parent: corda-5-alpha-tutorials-deploy
+    identifier: corda-5-alpha-tutorial-deploy-cordapp-packaging
+    weight: 2000
 section_menu: corda-5-alpha
 ---
 
-This section describes how to package your CorDapp as CPKs, CPBs, and CPIs. You can read more about the CorDapp packaging format in the [Key concepts](../../introduction/key-concepts.html#packaging) section.
+This section describes how to package your CorDapp as CPKs, CPBs, and CPIs. You can read more about the CorDapp packaging format in the [Key Concepts](../../introduction/key-concepts.html#packaging) section.
```
Comment (colin-hillis) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md:12]: Maybe define these acronyms the first time that we use them in a doc.

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md [hunk 11]
```diff
@@ -103,16 +109,19 @@ The gradle plugin builds the CPB. To create a CPB for a CPI:
...
     ```
 3. Import the signing key into Corda:
     ```shell
-    curl --insecure -u admin:admin -X PUT -F alias="signingkey1-2022" -F certificate=@signingkey1.pem https://localhost:8888/api/v1/certificates/codesigner
+    curl --insecure -u admin:admin -X PUT -F alias="signingkey1-2022" -F certificate=@signingkey1.pem https://localhost:8888/api/v1/certificates/cluster/code-signer
     ```
-{{< note >}}Use an alias that will be unique over time. Consider how certificate expiry will require new certificates with the same x500 name as existing certificates and define a naming convention that covers that use case.{{< /note >}}
+
+{{< note >}}
+Use an alias that will be unique over time. Consider how certificate expiry will require new certificates with the same X.500 name as existing certificates, and define a naming convention that covers that use case.
```
Comment (colin-hillis) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/cordapp-packaging.md:126]: Consider how certificate expiry will require new certificates with the same X.500 name as existing certificates, and define a naming convention that covers that use case.
>
After consideration that certificate expiry will require new certificates with the same X.500 name as existing certificates, define a suitable naming convention that covers that use case.

**Review by colin-hillis — CHANGES_REQUESTED** [review]
Thanks Amie

**Review by nadinequinr3 — APPROVED** [review]

**Review by yarivus — APPROVED** [review]
LGTM

**Review by colin-hillis — APPROVED** [review]
LGTM

**Pull request merged**